solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -105,7 +107,17 @@ public int hashCode() {
     result = (prime * result) + ((meta == null) ? 0 : meta.hashCode());
     result = (prime * result)
         + ((originalQuery == null) ? 0 : originalQuery.hashCode());
-    result = (prime * result) + ((efi == null) ? 0 : efi.hashCode());
+    if (efi == null) {
+      result = (prime * result) + 0;
+    }
+    else {
+      for (final Entry<String,String[]> entry : efi.entrySet()) {
+        final String key = entry.getKey();
+        final String[] values = entry.getValue();
+        result = (prime * result) + key.hashCode();
```

Comment:
The hashCode should be the same for efi's passed in different orders
hashCode({efi.user_query=foo efi.queryIntent=company}) == hashCode({efi.queryIntent=company efi.user_query=foo})

Add a unit test to verify